### PR TITLE
Feat: refactor transcode job

### PR DESCRIPTION
**Pull request by carshadi**
This PR deprecates the `transcode_job` module in favor of the new `zarr_upload_job.ZarrUploadJob` class, which extends `basic_job.BasicJob`.  

Job configs can be defined in a .csv file similarly to `BasicJob`. `ModalityConfig.extra_configs` is used to specify a path to a .yaml file defining the fields of `ZarrConversionConfigs`. If this path is `None`, sensible defaults are used.

### CODE REVIEW

File: src/aind_data_transfer/jobs/zarr_upload_job.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        0,
+        description="Default minimum of the neuroglancer display range. Default is 0."
+    )
+    ng_vmax: Optional[float] = Field(
+        200.0,
+        description="Default maximum of the neuroglancer display range. Default is 200."
+    )
+
+    @classmethod
+    def from_yaml(cls, yaml_path: Path):
```
Comment (jtyoung84) [inline src/aind_data_transfer/jobs/zarr_upload_job.py:78]: This is probably okay as it is and I've probably done things that way in the past. With pydantic BaseSettings, there is a way to define a sequence of resolvers including a config file inside a customize_sources method. I do that here: https://github.com/AllenNeuralDynamics/aind-codeocean-api/blob/377c475d9084bf9802a2e4effc8063a132e48ce9/src/aind_codeocean_api/credentials.py#L136

**Review by jtyoung84 — APPROVED** [review]
There are few warnings about adding new lines at the end of files, but it sees fine to me

**Comment by carshadi** [comment]
@cbrrrry  I had to make some changes to the output file names to account for the `aind-data-schema` update. Might be worth testing the diSPIM (now called HCR) upload one more time to see if everything still works on your end.

**Comment by carshadi** [comment]
Not clear why tests are failing

**Pull request merged**